src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -24,7 +24,7 @@ namespace NuGet.PackageManagement.UI
     /// The base class of PackageDetailControlModel and PackageSolutionDetailControlModel.
     /// When user selects an action, this triggers version list update.
     /// </summary>
-    public abstract class DetailControlModel : INotifyPropertyChanged, IDisposable
+    public abstract class DetailControlModel : TabViewModelBase, IDisposable
```

Comment:
Should this really be a Tab VM? I thought tabs were part of the detail control, not representing the entire control.

Reply:
```xml
<ContentControl Content="{Binding}">
  <ContentControl.Resources>
    <DataTemplate DataType="{x:Type nuget:DetailControlModel}">
      <nuget:PackageMetadataControl
        AutomationProperties.Name="{x:Static nuget:Resources.Accessibility_PackageMetadata}"
        DataContext="{Binding}"
        Margin="0,12,0,0"/>
    </DataTemplate>
    <DataTemplate DataType="{x:Type vm:ReadmePreviewViewModel}">
      <nuget:PackageReadmeControl
        DataContext="{Binding}"
        x:Name="_packageReadmeControl"
        AutomationProperties.Name="{x:Static nuget:Resources.Accessibility_PackageMetadata}"/>
    </DataTemplate>
  </ContentControl.Resources>
</ContentControl>
```

So from this code snippet we use the type of the VM to determine which view to render.

```xml
      <DataTemplate>
        <TextBlock Text="{Binding Header}" />
        <DataTemplate.Triggers>
          <DataTrigger Binding="{Binding Visible}" Value="False">
            <Setter Property="Visibility" Value="Collapsed" />
          </DataTrigger>
        </DataTemplate.Triggers>
      </DataTemplate>
```
In this snippet we use the `IsVisible` and `Header` elements in the TabViewModelBase class. I don't think an interface for the tab viewmodel makes sense since we would have to implement the notifyproperty change in all the implementations. 

Not sure if this works but I could add `ContentViewModel` to TabViewModelBase and maybe do this?
```xml
<ContentControl Content="{Binding ContentViewModel}">
  <ContentControl.Resources>
    <DataTemplate DataType="{x:Type nuget:DetailControlModel}">
      <nuget:PackageMetadataControl
        AutomationProperties.Name="{x:Static nuget:Resources.Accessibility_PackageMetadata}"
        DataContext="{Binding}"
        Margin="0,12,0,0"/>
    </DataTemplate>
    <DataTemplate DataType="{x:Type vm:ReadmePreviewViewModel}">
      <nuget:PackageReadmeControl
        DataContext="{Binding}"
        x:Name="_packageReadmeControl"
        AutomationProperties.Name="{x:Static nuget:Resources.Accessibility_PackageMetadata}"/>
    </DataTemplate>
  </ContentControl.Resources>
</ContentControl>
```

Reply:
So I discussed this a bunch with @jebriede  yesterday. I like the idea of using the VMs to determine which tabs are visible, however the DetailControlModel VM such a big VM that's handling so many different responsibilities that I think breaking it down to do this correctly would be so much work that I'm going to create a tech debt item to handle it in the future.

https://github.com/NuGet/Client.Engineering/issues/3054

Reply:
I am going to rename the TabViewModelBase though so it doesn't imply the VM is a tab